account_invoice_discount_lines/views/account_invoice.xml

```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="invoice_supplier_form" model="ir.ui.view">
+        <field name="name">Add applied discount field to invoice line in supplier invoice</field>
+        <field name="model">account.invoice</field>
+        <field name="inherit_id" ref="account.invoice_supplier_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='invoice_line_ids']/tree/field[@name='discount']"
+                   position="before">
+                <field name="discount" invisible="1"/>
+            </xpath>
+            <xpath expr="//field[@name='invoice_line_ids']/tree/field[@name='discount']"
+                   position="attributes">
+                <attribute name="string">Discount (%)</attribute>
+                <attribute name="name">discount_display</attribute>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="invoice_form" model="ir.ui.view">
+        <field name="name">Add applied discount field to invoice line in customer invoice</field>
+        <field name="model">account.invoice</field>
+        <field name="inherit_id" ref="account.invoice_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='invoice_line_ids']/tree/field[@name='discount']"
+                   position="before">
+                <field name="discount" invisible="1"/>
```

Comment:
How ever stupid this looks it works. I cannot find an explanation. 
This will not work

```
            <xpath expr="//field[@name='invoice_line_ids']/tree/field[@name='discount']"
                   position="after">
                <field name="discount_display" string="Discount (%)"/>
            </xpath>
            <xpath expr="//field[@name='invoice_line_ids']/tree/field[@name='discount']"
                   position="attributes">
                <attribute name="invisible">True</attribute>
            </xpath>
```

Reply:
That's very strange indeed. Usually, having the same field in the view twice can lead to problems. How does it not work to set the attribute this way? What is the (undesired) effect?